src/data_api/events.se

```diff
@@ -1,346 +1,420 @@
 
-def getEventPushedUp(event):
-    return(self.Events[event].pushedUp)
-
-def getResolutionAddress(event):
-    return(self.Events[event].resolutionAddress)
-
-def setEventPushedUp(event, val):
-    self.controller.assertIsWhitelisted(msg.sender)
-    self.Events[event].pushedUp = val
-    return(1)
+def isEventFinalized(eventId):
+    if (self.events[eventId].finalOutcome != 0):
+        return(1)
 
-# @return fxp
-def getForkOutcome(event):
-    whitelist = self.controller.assertIsWhitelisted(msg.sender)
-    if(msg.sender == INFO.getCreator(event) or whitelist):
-        return(self.Events[event].forkOutcome)
-    else:
-        log(type = logOutcome, event, self.Events[event].forkOutcome)
-
-def setForkOutcome(event, value):
-    self.controller.assertIsWhitelisted(msg.sender)
-    self.Events[event].forkOutcome = value
+def migrateEventThroughAllForks(eventId):
+    if (self.isEventFinalized(eventId)):
+        return(1)
+    eventStillMigrating = 1
+    while (eventStillMigrating):
+        # this will loop until we run out of gas, follow forking events until there are no more, or have reached an active fork
+        eventStillMigrating = self.migrateEventThroughOneFork(eventId)
     return(1)
 
-# @return fxp
-def getForkEthicality(event):
-    return(self.Events[event].forkEthicality)
-
-def setForkEthicality(event, value):
-    self.controller.assertIsWhitelisted(msg.sender)
-    self.Events[event].forkEthicality = value
+# returns 0 if no move occurs, 1 if move occurred, throws if a fork not yet resolved
+def migrateEventThroughOneFork(eventId):
+    if (self.isEventFinalized(eventId)):
+        return(0)
+    targetBranch = self.events[eventId].branch
+    forkingEventId = self.controller.lookup('branches').getForkingEvent(branch)
+    # no fork, no need to move
+    if (forkingEventId == 0):
+        return(0)
+    # this event is the forking event
+    if (forkingEventId == eventId):
+        return(0)
+    # fork not yet resolved, can't move event
+    if (self.events[forkingEvent].finalOutcome == 0):
+        ~invalid()
+    # pay back bond holders
+    if (self.events[eventId].limitedReportersDisputeBondHolder != 0):
+        self.events[eventId].limitedReportersDisputeBondHolder = 0
+        # self.controller.lookup('reputationToken').transfer(self.events[eventId].limitedReportersDisputeBondHolder, self.limitedReportersDisputeBondAmount)
+    if (self.events[eventId].allReportersDisputeBondHolder != 0):
+        self.events[eventId].allReportersDisputeBondHolder = 0
+        # self.controller.lookup('reputationToken').transfer(self.events[eventId].allReportersDisputeBondHolder, self.allReportersDisputeBondAmount)
+    # follow the forking event to its branch
+    self.events[eventId].branch = self.events[forkingEventId].branch
+    # reset reporting state back to un-reported (except for automated)
+    self.events[eventId].reportingWindow = self.controller.lookup('reporting').getCurrentReportingWindow() + 1
```

Comment:
This is correct if the event's reporting window was in the past / the event should expire asap, but if it expires in the future make it the same future window as before [so no update].

so if reportingWindow < currentWindow + 1, update it to currentWindow + 1

otherwise don't